### PR TITLE
Add an explicit treatment input argument to causaltree/forest

### PR DESCRIPTION
**Pull request by jeongyoonlee**
## Proposed changes

This PR adds an explicit `treatment` input argument to `CausalTreeRegressor.fit()` and `CausalRandomForestRegressor.fit()`. 

Originally, it used `sample_weight` for treatment assignments to use the `sklearn.tree` APIs without modification, but, with #686, we don't use `sklearn.tree` anymore, and we're free to modify inputs to `CausalTreeRegressor` and `CausalRandomForestRegressor`.

All tests and notebook examples for causal tree/forest ran successfully without errors.

## Types of changes

What types of changes does your code introduce to **CausalML**?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/uber/causalml/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/uber/causalml)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

This change will make it easy to add multiple treatment support to causal tree/forest (#775).

### CODE REVIEW

File: causalml/inference/tree/causal/_tree.py [hunk 1]
```diff
@@ -42,7 +42,13 @@ def _support_missing_values(self, X) -> bool:
         return False
 
     def fit(
-        self, X, y, sample_weight=None, check_input=True, X_idx_sorted="deprecated"
+        self,
+        X,
+        y,
```
Comment (paullo0106) [inline causalml/inference/tree/causal/_tree.py:47]: @jeongyoonlee do we want to swap the order for `y` and `treatment` for consistency and also i see another lines are calling this function passing in the arguments in a different order? `tree.fit(X, treatment, y, sample_weight=curr_sample_weight, check_input=False)`

Comment (jeongyoonlee) [reply causalml/inference/tree/causal/_tree.py:47]: Thanks, Paul. Good catch! I will update it with the fix.

File: causalml/inference/tree/causal/causaltree.py [hunk 2]
```diff
@@ -153,19 +152,19 @@ def __init__(
...
-            X: : (np.ndarray), feature matrix
-            y: : (np.ndarray), outcome vector
-            treatment: : (np.ndarray), treatment vector
-            sample_weight: (np.ndarray), sample_weight
-            check_input: (bool)
+            X (np.ndarray): feature matrix
+            treatment (np.ndarray): treatment vector
+            y (np.ndarray): outcome vector
+            sample_weight (np.ndarray): sample_weight
+            check_input (bool, optional), default=False
```
Comment (paullo0106) [inline causalml/inference/tree/causal/causaltree.py:167]: nit
`check_input (bool, optional): default=False`

**Review by ras44 — COMMENTED** [review]
only general comment is to perhaps consider a "major version" bump to 0.16 once this is merged given that the API has changed and treatment is now a required arg

**Review by paullo0106 — APPROVED** [review]

**Comment by jeongyoonlee** [comment]
@alexander-pv, can you take a look at this PR before I merge it? Thanks in advance!

**Pull request merged**